src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp

```diff
@@ -117,6 +117,13 @@
           "max heap size.")                                                 \
           range(0,100)                                                      \
                                                                             \
+  product(uintx, ShenandoahOldMinFreeThreshold, 5, EXPERIMENTAL,            \
+          "Percentage of free heap memory below which most heuristics "     \
+          "old generation trigger collection independent of other   "       \
```

Comment:
"Percentage of free old generation memory below which most heuristics will trigger its collection independent of other ..."

Similarly, I'd modify the documentation of "ShenandoahMinFreeThreshold" to state:
"Percentage of free heap memory (or young generation memory, in
 generational mode) below which most heuristics trigger its collection independent of ..."

ASIDE: I was thinking that these may be worth making "manageable" in the future, but I realize now that there is not precedent for that, e.g. in the case of InitiatingHeapOccupancyPercent. I do wonder why not though, since it can allow for some flexibility and external dynamic adaptation of the VM's settings.

Reply:
I agree with the suggestions for the documentation changes.

The `SoftMaxHeapSize` is "manageable" and that is a factor when checking these thresholds. However, the soft capacity cannot be directly configured for each generation (it is calculated based on `NewSize`, `NewRatio`).

Reply:
Updating docs based on @ysramakrishna's suggestions. Its not exactly word for word.

> Percentage of free old generation memory below which most heuristics will trigger its collection independent of other ..

I did not include the `will`.